fgcm/fgcmGray.py

```diff
@@ -1072,23 +1128,31 @@ def makeExpGrayPlots(self):
                                   self.expGrayCheckDeltaT * 24 * 60))
                 continue
 
-            fig=plt.figure(1,figsize=(8,6))
+            fig = makeFigure(figsize=(8, 6))
             fig.clf()
 
             ax=fig.add_subplot(111)
 
             ax.hexbin(expGray[use0[matchInd[ok]]]*1000.0,
-                      expGray[use1[ok]]*1000.0, bins='log')
+                      expGray[use1[ok]]*1000.0, bins='log', cmap=colormaps.get_cmap("viridis"))
             ax.set_xlabel('EXP_GRAY (%s) (mmag)' % (self.fgcmPars.bands[bandIndex0]))
             ax.set_ylabel('EXP_GRAY (%s) (mmag)' % (self.fgcmPars.bands[bandIndex1]))
             ax.plot([-0.01 * 1000, 0.01 * 1000],[-0.01 * 1000, 0.01 * 1000],'r--')
 
             if self.plotPath is not None:
-                fig.savefig('%s/%s_expgray-compare_%s_%s.png' % (self.plotPath,
-                                                                 self.outfileBaseWithCycle,
-                                                                 self.fgcmPars.bands[bandIndex0],
-                                                                 self.fgcmPars.bands[bandIndex1]))
-            plt.close(fig)
+                if self.butlerQC is not None:
```

Comment:
So (for better or worse) it's the same code that does the plotting as does the fit for the width which we want for logging and recording downstream.  Previously, this could potentially have bad side effects because of use of `matplotlib.pyplot`.  But now it doesn't have any side effects!  I've added comments to this effect.